tests/um_fields_subset_test.py

```diff
@@ -1,102 +1,110 @@
 
-import warnings
 import pytest
 from copy import deepcopy
 from unittest.mock import patch, MagicMock
-from hypothesis import HealthCheck, given, settings
+from hypothesis import HealthCheck, given, settings, strategies as st
 from hypothesis.extra import numpy as stnp
-from hypothesis import given, settings, strategies as st
-from um_fields_subset_mule import (parse_args, field_not_present_warning, include_fields, exclude_fields, filter_fieldsfile, create_default_outname, PROGNOSTIC_STASH_CODES, void_validation)
-from hypothesis import given, strategies as st
+from um_fields_subset_mule import (
+    parse_args, 
+    field_not_present_warning, 
+    include_fields, exclude_fields, 
+    filter_fieldsfile, 
+    create_default_outname, 
+    void_validation, 
+    convert_to_list, 
+    PROGNOSTIC_STASH_CODES,
+)
 import numpy as np
-from itertools import chain
-PROGNOSTIC_STASH_CODES = tuple(chain(range(1,999+1), range(33001,34999+1)))
 
-def test_convert_to_list_valid():
-    """
-    Test convert_to_list with valid input.
-    """
-    assert convert_to_list("1,2,3") == [1, 2, 3]
-    assert convert_to_list("10,20,30") == [10, 20, 30]
+@pytest.mark.parametrize(
+    "input, expected_output, should_raise",
+    [
+        ("1,2,3", [1, 2, 3], False),
+        ("10,  20,30  ", [10, 20, 30], False),
+        ("10.1,10,32", None, True),  # Contains a non-integer
+        ("10 20 30", None, True),    # Missing commas
+        ("-1,-2,-3", None, True),    # Contains negative numbers
+        ("0,1,2", None, True),        # Contains zero
+        ("a,1,2", None, True)        # Contains zero
```

Comment:
```suggestion
        ("a,1,2", None, True)        # Contains a non-number
```

I really like the comments!